### ISSUE TITLE
Android Realm - RealmBaseAdapter memory issue

### ISSUE THREAD

**Issue by acampayo** [realm/realm-java#1201]
I have a memory issue with my RealmBaseAdapter. The problem is that the memory increases when a new row of the ListView is created and when the system needs to release memory it seems it destroys my "comercios" RealmResults and the app crash when I scroll the list. I'm not very sure what is happening.

**My Adapter:**

```
public class ComercioAdapter extends RealmBaseAdapter<Comercio> implements ListAdapter {

    public ComercioAdapter(Context context, RealmResults<Comercio> comercios) {
        super(context, comercios, false);
    }

    public void reloadData(RealmResults<Comercio> comercios){
        this.realmResults = comercios;
        notifyDataSetChanged();
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        ViewHolder holder;

        if (view == null) {
            view = ((Activity) context).getLayoutInflater().inflate(R.layout.adapter_comercio, null);
            holder = new ViewHolder();
            holder.name = (TextView) view.findViewById(R.id.name);
            view.setTag(holder);
        } else {
            holder = (ViewHolder) view.getTag();
        }

        Comercio comercio = getItem(i);
        holder.name.setText(comercio.getName());
        return view;
    }

    private class ViewHolder {
        private TextView name;
    }

}
```
**Stack trace:**

`4406-4406/com.cmrlife.development A/libc﹕ Fatal signal 11 (SIGSEGV) at 0x00f600f2 (code=1), thread 4406`

I also have two activities which extends a BaseActivity activity where I create and close the Realm instance of each activity. 

**BaseActivity:**

```
public class BaseActivity extends AppCompatActivity {

    private Realm mRealm;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mRealm = Realm.getInstance(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mRealm.close();
    }

    public Realm getRealm() {
        return mRealm;
    }
}
```

**This is the structure of my app:**

**MainActivity (Extends BaseActivity)**

- Shows two fragments with a ListView each of them
- When System releases memory all works fine

**SecondActivity (Extends BaseActivity)**

- Shows a ListView of RealmObjects in a fragment
- When System releases memory the issue is reproduced

Could be a problem with the Realm instances?

**Comment by cmelchior** [comment]
Hi @acampayo 
There should be nothing inherently wrong with how your code is structured. Are you able to share your codebase or a test project that can reproduce this? If you want to do it privately you can send it to help@realm.io

**Comment by cmelchior** [comment]
Hi @acampayo  Thank you for the code. I don't seem to be able to make it crash from running out of memory. Do you have some steps to reproduce this?

**Comment by acampayo** [comment]
Hi Christian, thanks for your email. Yes to reproduce it you have to follow
this steps:

- Tap on "Cercanías" tab which is located at the first screen
- Select one of the items of the list (I recommend the last one)
- On the second scroll up and down the list until the app crash

If you check the memory status with Android Studio, you will see that the
memory is increasing while you're scrolling the list until the app crash.

If you need more information about it, don't hesitate to contact me.

Thanks!


2015-06-11 18:36 GMT+02:00 Christian Melchior <notifications@github.com>:

**Comment by cmelchior** [comment]
Hi @acampayo 
I cannot reproduce this. Is is correct that scrolling to the bottom of Cercanías and clicking "Ver mas vestuario y calzado" doesn't actually do anything and you manually have to swipe to the next page (Categorias)?

If that is correct It looks fine to me. Memory usage increases a bit, but when the garbage collector runs it is successfully reclaimed. What phone/emulator/android version are you testing on?

**Comment by cmelchior** [comment]
Was fixed by upgrading to 0.81.1.
